subprojects/base-services/src/integTest/groovy/org/gradle/internal/exceptions/DefaultMultiCauseExceptionIntegrationTest.groovy

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.exceptions
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class DefaultMultiCauseExceptionIntegrationTest  extends AbstractIntegrationSpec {
+    def 'when tasks throw exceptions that offer resolutions, those resolutions are included'() {
+        given:
+        buildFile << """
+            ${defineTestException()}
+
+            tasks.register('myTask') {
+                doLast {
+                    throw new TestResolutionProviderException('resolution1')
+                }
+            }
+        """
+
+        when:
+        fails 'myTask'
+
+        then:
+        assertHasSuggestions('resolution1')
```

Comment:
We have a test fixture to do this.

Take a look at https://github.com/gradle/gradle/blob/a939d6cd39d7126db1ff9949853e58dc2aa57c11/platforms/jvm/language-java/src/integTest/groovy/org/gradle/jvm/toolchain/JavaToolchainDownloadIntegrationTest.groovy#L59

This would become something like:
```
failure.assertHasDescription(???).assertHasCause(???).assertHasResolution("resolution1")
```
where ??? are not completely obvious to me without running this.

Reply:
OK.  I was looking for "suggestions" and didn't see anything.  I'll switch to using these methods.